From dfd32d3af36d45e0ab143a4756e5880e9fbd8ffd Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:29 +0900
Subject: [PATCH 0967/1083] Revert "drm/rcar-du: Don't re-reserve hardware
 plane at each update"

This reverts commit 77fb180458fde4a9ca7198984e62dfaa49ade4af.
(cherry picked from commit c178cc0ab0bcf15967bab7de65c1ba9854a518b0)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_plane.c | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 220ca00..110fd13 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -206,7 +206,6 @@ rcar_du_plane_update(struct drm_plane *plane, struct drm_crtc *crtc,
 	struct rcar_du_plane *rplane = to_rcar_plane(plane);
 	struct rcar_du_device *rcdu = plane->dev->dev_private;
 	const struct rcar_du_format_info *format;
-	unsigned int nplanes;
 	int ret;
 
 	format = rcar_du_format_info(fb->pixel_format);
@@ -221,8 +220,6 @@ rcar_du_plane_update(struct drm_plane *plane, struct drm_crtc *crtc,
 		return -EINVAL;
 	}
 
-	nplanes = rplane->format ? rplane->format->planes : 0;
-
 	rplane->crtc = crtc;
 	rplane->format = format;
 	rplane->pitch = fb->pitches[0];
@@ -234,15 +231,9 @@ rcar_du_plane_update(struct drm_plane *plane, struct drm_crtc *crtc,
 	rplane->width = crtc_w;
 	rplane->height = crtc_h;
 
-	/* Reallocate hardware planes if the number of required planes has
-	 * changed.
-	 */
-	if (format->planes != nplanes) {
-		rcar_du_plane_release(rplane);
-		ret = rcar_du_plane_reserve(rplane);
-		if (ret < 0)
-			return ret;
-	}
+	ret = rcar_du_plane_reserve(rplane);
+	if (ret < 0)
+		return ret;
 
 	rcar_du_plane_compute_base(rplane, fb);
 	rcar_du_plane_setup(rplane);
-- 
1.8.3.2

